iot_input/models/iot_device_input.py

```diff
@@ -63,6 +70,29 @@ def call_device(self, value):
             self._add_action_vals(value, res))
         return res
 
+    @api.model
+    def get_device_input(self, serial, passphrase, value):
+        device_input = self.with_context(
+            active_test=False).search([('serial', '=', serial)])
+        if not device_input:
+            return {'status': 'error',
+                    'message': _('Input with serial: {} cannot be found').
+                    format(serial)}
+        if not device_input.active:
+            return {'status': 'error',
+                    'message':
+                    _('Input with serial: {} is inactive, no data will be logged').
+                    format(serial)}
+        if device_input.passphrase != passphrase:
```

Comment:
I think you should never respond that, external users should never know if the passphrase or the serial are defined, this is just a matter of security. At the end, the first error (no input) is enough in this case for me

Reply:
True. I'll remove serial from the response and change to more generic message